src/main/java/com/mixpanel/android/mpmetrics/AnalyticsMessages.java

```diff
@@ -427,6 +427,10 @@ private void sendData(MPDbAdapter dbAdapter, MPDbAdapter.Table table, String[] u
                                 } catch (UnsupportedEncodingException e) {
                                     throw new RuntimeException("UTF not supported on this platform?", e);
                                 }
+                                if (mFailedRetries > 0) {
+                                    mFailedRetries = 0;
+                                    removeMessages(FLUSH_QUEUE);
```

Comment:
Ah, okay. Sounds good.
I'm looking forward to the followup PR that rips out all of this delayed-job stuff and replaces it with a background thread a la the iOS library. ;)